gapic-generator/templates/default/service/client/_client.erb

```diff
@@ -55,60 +47,60 @@ class <%= service.client_name %>
   #   `GRPC::Core::CallCredentials` object.
   #   A `Proc` will be used as an updater_proc for the Grpc channel. The proc
   #   transforms the metadata for requests, generally, to give OAuth credentials.
-  # @param scope [String, Array<String>]
-  #   The OAuth scope (or scopes) for this service. This parameter is ignored if
-  #   an updater_proc is supplied.
-  # @param timeout [Numeric]
-  #   The default timeout, in seconds, for calls made through this client.
-  # @param metadata [Hash]
-  #   Default metadata to be sent with each request. This can be overridden on a
-  #   per call basis.
+  # @param config [Google::Gax::Configuration]
+  #   The configuration object to use in place of the default configuration. It is
```

Comment:
We don't have a way to express "use the default configuration except for this one changed field" do we?

Reply:
We do not. Right now we implement a `create` and a `wrap` method that does this, but this is something that we could add directly to `Google::Gax::Configuration`, to make it more clear.

Reply:
This PR has been updated to use `Configuration#derive!` from `googleapis/gax-ruby#195. This allows you to "inherit" from a configuration and change values, while inheriting all other values from the parent config object.

The instance config (`Client#configure`) is derived from the global config (`Client.configure`).

Reply:
So the expected usage is something like this?

```ruby
Client.new(config: Client.configure.derive! {|c| c.timeout = 100})
```

That still feels highly clunky, especially since it's replacing

```ruby
Client.new timeout: 100
```

Reply:
That would work, but I think this usage is more likely:

```ruby
Client.new { |c| c.timeout = 100 }
```

Or:

```ruby
client = Client.new
client.configure.timeout = 100
```

Reply:
Oh. Okay that makes sense, but it does leave the question of whether the derive stuff is even necessary.

Reply:
In this instance, we have a global config, and a per-instance config. The global config would not be derived, but the per-instance config would be.

```ruby
Client.configure.derived? #=> false
Client.new.configure.derived? #=> true
```

The difference here is that fields can be added/removed in the global config, but only values can be changes in the per-instance config.

Reply:
Ah, okay. 👍🏼

Reply:
Looking at this again, the original assumption here was that a config object could be passed in so that the same config object could be used between a service client and an operations client. However, that ended up not working because they have different methods hence the need for separate configs. So I am going to remove the constructor argument. Users should either configure using the constructor's block or by calling the `#configure` instance method.

Reply:
Removed the argument. Users should use the closure or `#configure` instance method to customize.